src/operator/z1.rs

```diff
@@ -181,6 +329,42 @@ where
     }
 }
 
+impl<T> UnaryOperator<T, T> for Z1Nested<T>
+where
+    T: Clone + 'static,
+{
+    fn eval(&mut self, i: &T) -> T {
+        assert!(self.timestamp <= self.val.len());
```

Comment:
These asserts could probably be `debug_assert!()`s since it's not required for safety and it's more of a "keep us from writing buggy code" kind of thing